designs/virtualization/4.6/Add-os-image-to-common-template/add-os-image-to-common-template 4.6.md

```diff
@@ -0,0 +1,56 @@
+---
+parent: Virtualization
+version: 4.6
+---
+# Add an operating system image to a common template
+
+## About
+
+We strive to provide an off-the-shelf OS image with every common template.
+This an ongoing effort.
+Hopefully this will provide the user with a much easier experience to creating a virtual machine.
+This OS image takes the shape of a disk image(data-volume) and can be attached to the OS template (one of the parameters within the common templates).
```

Comment:
one thing we'll need to think about is when common templates ship with an image already. This feature maybe become less useful or maybe they can add multiple sources for a single OS depending on the boot source type. 
It could also be for cases where we don't ship a a boot source with a common template they can see that they can add the source themselves (Likely for Windows VMs)

Reply:
I agree, and I believe this design is an "in-between" solution that will soon not be the main path.

If we go with adding an "Add source" from the VM templates list view or details page, we still need to cover what happens when the user clicks on "Create a VM (from a template without image)".
One alternative to this design would be to pop up an "add source/image" modal as a pre-requisite to a VM template with a missing source/image.
It depends on if we want to provide the ability to create a VM from a template with a "one-time" image. If we do, then we still need to provide a Create VM action fro templates with missing images.

When common templates shipped with an image, then the UI would be that this OS lists as "with image" and will have the checkbox to clone (selected by default). This is already covered in the design.

Reply:
maybe for now we should just add a design that covers the scenario when a user tries to upload a new image for an OS that already has one.

Reply:
Doesn't the last image cover this? (line 55)